python/katana/local/edge_iterator.py

```diff
@@ -0,0 +1,24 @@
+class EdgeIterator:
+    """
+    An iterator over the edges of a graph
+    """
+
+    def __init__(self, graph):
+        self._graph = graph
+        self._curr_node = 0
+        self._curr_edge = 0
+
+    def __next__(self):
+        if self._curr_edge >= self._graph.num_edges():
+            raise StopIteration
+        if len(self._graph.edges(self._curr_node)) == 0:
+            self._curr_node += 1
+            return self.__next__()
+        if self._curr_edge > self._graph.edges(self._curr_node)[-1]:
+            self._curr_node += 1
+        next_edge = (self._curr_node, self._graph.get_edge_dest(self._curr_edge))
```

Comment:
Using an unadorned tuple encourages bad client code. The reason is there is no way for the client to be explicit about what they actually mean when accessing the values in the pair. It's much better to use a custom class or named tuple.